gamemodes/terrortown/gamemode/server/sv_corpse.lua

```diff
@@ -235,19 +235,36 @@ local function ttt_call_detective(ply, cmd, args)
 
 	if IsValid(rag) and rag:GetPos():Distance(ply:GetPos()) < 128 then
 		if CORPSE.GetFound(rag, false) then
+			local plyTable = GetRoleChatFilter(ROLE_DETECTIVE, true)
+
+			hook.Run("TTT2ModifyCorpseCallRadarReceipients", plyTable, rag, ply)
+
 			-- show indicator in radar to detectives
 			net.Start("TTT_CorpseCall")
 			net.WriteVector(rag:GetPos())
-			net.Send(GetRoleChatFilter(ROLE_DETECTIVE, true))
+			net.Send(plyTable)
 
-			LANG.Msg("body_call", {player = ply:Nick(), victim = CORPSE.GetPlayerNick(rag, "someone")}, MSG_CHAT_PLAIN)
+			LANG.MsgAll("body_call", {player = ply:Nick(), victim = CORPSE.GetPlayerNick(rag, "someone")}, MSG_CHAT_PLAIN)
 		else
 			LANG.Msg(ply, "body_call_error")
 		end
 	end
 end
 concommand.Add("ttt_call_detective", ttt_call_detective)
 
+---
+-- This hook is called after a players pressed the "call detective" button and
+-- all requirements were met. It can be used to modify the table of players that
+-- should receive the corpse radar ping.
+-- @param Player notifiedPlayers The table of players that will be notified
+-- @param Entity ragdoll The ragdoll whose coordinates are about to be sent
+-- @param Player ply The player that pressed the "call detective" button
+-- @hook
+-- @realm server
+function GM:TTT2ModifyCorpseCallRadarReceipients(nofifiedPlayers, ragdoll, ply)
```

Comment:
`nofifiedPlayers` should be `notifiedPlayers`

Reply:
Oh great, thanks. I'll rework it in a minute and let you know. Are you blackmagicfire on discord?

Reply:
Yup, I am blackmagicfine.

Reply:
So, I thought about this for a while. At first I thought your suggestion might be a problem because it can be weird with role changes.
However I came to the conclusion that it doesn't matter. In the worst case, you see a corpse position for up to 45 seconds after your role change which is no problem in my opinion. Thanks for your input!

Reply:
This anyways isn't an item issue. It depends on the role, if the radar was bought and the role changed, the radar should be kept so it can still be used. If the radar is included in a role and that role got changed, then the old role has to handle this on it's own (removing the item in the right hook).

Reply:
I'm not that deep into the code, but is it really based on the radar? To me it looks like the code happens to be in the radar file, but has nothing to do with the actual radar. This fact is supported by the situation where a detective has a corpse point in world without having a radar.